### PR TITLE
feature: support xid load balance

### LINKED ISSUE #4688: supports load balance by xid
## Why you need it?
 Is your feature request related to a problem? Please describe in details

Currently Seata interacts with TC load balance for each time, which causes some problems:

1. XA 2phase branch commit/rollback could not find the original XAConn

2. Atomic state issues between nodes, such as missing later registered branch transactions when rolling back

3. Log aggregation for a single transaction

4. Metrics single-node active transaction negative number problem

Therefore, load balance needs to be done according to xid to solve the above problems. In addition, the availability of TC should also be considered.

## How it could be?
A clear and concise description of what you want to happen. You can explain more about input of the feature, and output of it.

### PR DESCRIPTION
**Pull request by a364176773**
- [ ] I have registered the PR [changes](https://github.com/seata/seata/tree/develop/changes).

### Ⅱ. Does this pull request fix one issue?

fixes #4688

### CODE REVIEW

File: discovery/seata-discovery-core/src/main/java/io/seata/discovery/loadbalance/XIDLoadBalance.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+        LoadBalanceFactory.RANDOM_LOAD_BALANCE);
+
+    private static final String SPLIT = ":";
+
+    @Override
+    public <T> T select(List<T> invokers, String xid) throws Exception {
+        if (StringUtils.isNotBlank(xid) && xid.contains(SPLIT)) {
+            String[] xidArray = xid.split(SPLIT);
+            int port = Integer.parseInt(xidArray[1]);
+            String ip = xidArray[0];
```
Comment (slievrly) [inline discovery/seata-discovery-core/src/main/java/io/seata/discovery/loadbalance/XIDLoadBalance.java:46]: consider ipv6 support

Comment (a364176773) [reply discovery/seata-discovery-core/src/main/java/io/seata/discovery/loadbalance/XIDLoadBalance.java]: done

**Comment by a364176773** [comment]
> https://github.com/seata/seata/pull/4713/files#diff-67c903540585b858205a835c0483c7d97511061428c5aaf2de0f70a774579ee1R59 expect to downgrade to other load balancing modes to ensure availability.

done

**Review by slievrly — APPROVED** [review]
LGTM

**Pull request merged**